### PR TITLE
re-doing Docker Engine overview topics for v.1.12

### PR DESCRIPTION
**Pull request by londoncalling**
**\- What I did**

**\- How I did it**

**\- How to verify it**

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Victoria Bialas victoria.bialas@docker.com

### CODE REVIEW

**Comment by londoncalling** [comment]
@SvenDowideit  please take a look, this is the start of re-doing the Docker Engine overview. Will update this file more with current content, but this is a start of it.

File: docs/understanding-docker.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
 <!--[metadata]>
 +++
-aliases = ["/introduction/understanding-docker/"]
-title = "Understand the architecture"
+aliases = [
+"/introduction/understanding-docker/",
+"/engine/userguide/basics/",
+"/engine/quickstart.md"
```
Comment (thaJeztah) [inline docs/understanding-docker.md]: think this should be `"/engine/quickstart/"` (without `.md`)

File: docs/understanding-docker.md [hunk 2]
```diff
@@ -37,6 +43,24 @@ and testing
...
+
+* A REST API which specifies interfaces that programs can use to talk to the daemon and instruct it what to do.
+
+* A command line interface (CLI) client.
+
+![Docker Engine Components Flow](article-img/engine-components-flow.png)
+
+The CLI imakes use of the Docker REST API to control or interact with the Docker daemon through scripting or direct CLI commands. Many other Docker applications make use of the underlying API and CLI.
+
+The daemon creates and manages Docker objects.  Docker objects include images, containers, networks, data volumes, and so forth.
```
Comment (thaJeztah) [inline docs/understanding-docker.md]: Can you wrap to 80 chars? (This line, and the one above)

**Comment by thaJeztah** [comment]
CI failures are genuine; looks like there are some links to this page broken by this;

```
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.23.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.20.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.18.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.22.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.21.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.24.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 * link error: (in page engine/reference/api/docker_remote_api_v1.19.md) ../../quickstart.md#bind-docker-to-another-host-port-or-a-unix-socket
00:50:31 
```

**Comment by justincormack** [comment]
Should we update the top level `README.md` as well to match? This has struck me as a bit dated for a while.

**Comment by londoncalling** [comment]
@SvenDowideit @thaJeztah @justincormack All the broken links are pointing to one particular subtopic in `quickstart.md`. I do have `quickstart.md` re-directed to `understanding-docker.md`, but that won't resolve these more specific links. The (now) missing subtopic is the one you and I looked at, Sven: https://docs.docker.com/engine/quickstart/#bind-docker-to-another-host-port-or-a-unix-socket.

I can fix this by moving this one topic somewhere else, and then updating those links to its new location, but I'm not sure where to elegantly add it in. (My knowledge of networking is not up to this task right now.) The topics I've looked at where it _might_ fit are:

- User Guide "network containers" example: https://docs.docker.com/engine/userguide/containers/networkingcontainers/

- User Guide "network configuration" topic, somewhere in there ( first topic is here: https://docs.docker.com/engine/userguide/networking/ and out of date index page for it is here: https://docs.docker.com/engine/userguide/)

Do either of those places make sense? I'd welcome some ideas on integrating this topic into a place where it makes sense, and other intro text to to describe context for it. It seems both legacy and still relevant.

Also, as a very important aside, when I was browsing through network topics, I noticed references to examples that are "dependent on Virtual Box being installed" (e.g., https://docs-preview.docker.com/engine/userguide/networking/get-started-overlay/#prerequisites) . I imagine there are a lot of these peppered throughout the docs. How do we handle this right now for v.1.12 when Piñata will be on the scene?

**Comment by thaJeztah** [comment]
@londoncalling the redirects are only useful for people that have bookmarked the old location, but don't fix internal links (because Hugo will look for a file named `quickstart.md` to actually *generate* the link. changing those links to point to `understanding-docker.md` should fix the failures.

w.r.t. VirtualBox; I suggest to do a follow-up PR to clean those up; but there's quite some examples using VirtualBox / Docker ToolBox. Although we should fix those, I think it's "ok" to do so in the coming weeks, ToolBox is not entirely "gone" yet (but just my 0.02c)

**Comment by londoncalling** [comment]
@thaJeztah understand, regarding the re-directs, but the problem is "understanding docker" doesn't cover that use case that the API topics are pointing to. We saw it as way to low level for an overview to Docker, so I need to move that particular topic to somewhere that makes sense, then change the API links to there. 

Agreed that we can't tackle the whole VirtualBox thing right now :-) soon though. I just wanted to call it out for myself and the rest of us.

**Comment by thaJeztah** [comment]
@londoncalling oh! get it; I think that section could be moved to the dockerd.md document; https://docs.docker.com/engine/reference/commandline/daemon/ (or now; https://github.com/docker/docker/blob/master/docs/reference/commandline/dockerd.md) would that make sense?

Possibly adding links to https://docs.docker.com/engine/security/https/ and https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface for the security aspect of doing so

**Comment by thaJeztah** [comment]
Or, alternatively in the "main" remove-API document; https://docs.docker.com/engine/reference/api/docker_remote_api/, as it's related to the remote API

**Comment by londoncalling** [comment]
@thaJeztah @justincormack actually, from what I can tell, this covers it  file you pointed me to, wouldn't you say so? [Daemon socket option ](https://docs.docker.com/engine/reference/commandline/daemon/#daemon-socket-option)

I'm going to re-direct those links to this topic and save off the original topic ([Bind Docker to another host/port or a Unix socket](https://docs.docker.com/engine/quickstart/#bind-a-service-on-a-tcp-port) ) to my desktop along with everything else in the quickstart, just in case we find something else isn't covered later. Thanks for pointing me to the daemon reference, @thaJeztah

**Comment by londoncalling** [comment]
@justincormack okay I'm not sure which README you're talking about now. But I've made the changes to the files in a way that should make sense. Please review, then let me know more specifically what README.

**Comment by SvenDowideit** [comment]
I plan on working through it later today - if things goto plan :)

**Comment by londoncalling** [comment]
@thaJeztah @SvenDowideit I'm going to merge this so that the broken links will be resolved in the docs builds

**Comment by londoncalling** [comment]
@thaJeztah can you merge this PR for me? It's blocking for other work we have to do.

**Comment by thaJeztah** [comment]
@londoncalling oh, yes! LGTM

**Pull request merged**